cs1/tokenbucket.py

```diff
@@ -0,0 +1,29 @@
+import time
+
+
+class TokenBucket:
+
+    def __init__(self, tokens, time_unit):
+        self.tokens = tokens
+        self.time_unit = time_unit
+        self.bucket = tokens
+        self.last_check = time.time()
+
+    def consume(self):
+        if (self.bucket < 1):
+            time.wait(self.time_unit / self.tokens)
+            self.bucket = self.bucket + 1
```

Comment:
Shouldn't this be subtracting? I don't see how this will hit ever decrease if initially `bucket == tokens`: this adds one, then the if statement below caps at `tokens`.

Please test the code.